protocols/gossipsub/src/behaviour.rs

```diff
@@ -228,7 +229,7 @@ pub struct Gossipsub<
 
     /// A map of peers to their protocol kind. This is to identify different kinds of gossipsub
     /// peers.
-    peer_protocols: HashMap<PeerId, PeerKind>,
+    peer_protocols: HashMap<PeerId, PeerConnections>,
```

Comment:
Both key and doc comment are off now, right?

How about:
```suggestion
    /// A set of connected peers, indexed by their [`PeerId`], tracking both the [`PeerKind`] and the set of [`ConnectionId`]s.
    connected_peers: HashMap<PeerId, PeerConnections>,
```

Reply:
I still find the _key_ confusing and would prefer `connected_peers`. @AgeManning any particular reason why keeping `peer_protocols`?